config/config_mgmt.py

```diff
@@ -0,0 +1,864 @@
+            print(e)
+            raise e
+
+        return True
+
+    def breakOutPort(self, delPorts=list(), addPorts= list(), portJson=dict(), \
+        force=False, loadDefConfig=True):
+        '''
+        This is the main function for port breakout. Exposed to caller.
+
+        Parameters:
+            delPorts (list): ports to be deleted.
+            addPorts (list): ports to be added.
+            portJson (dict): Config DB json Part of all Ports, generated from
+                platform.json.
+            force (bool): if false return dependecies, else delete dependencies.
+            loadDefConfig: If loadDefConfig, add default config for ports as well.
+
+        Returns:
+            (deps, ret) (tuple)[list, bool]: dependecies and success/failure.
+        '''
+        MAX_WAIT = 60
+        try:
+            # delete Port and get the Config diff, deps and True/False
+            delConfigToLoad, deps, ret = self._deletePorts(ports=delPorts, \
+                force=force)
+            # return dependencies if delete port fails
+            if ret == False:
+                return deps, ret
+
+            # add Ports and get the config diff and True/False
+            addConfigtoLoad, ret = self._addPorts(ports=addPorts, \
+                portJson=portJson, loadDefConfig=loadDefConfig)
+            # return if ret is False, Great thing, no change is done in Config
+            if ret == False:
+                return None, ret
+
+            # Save Port OIDs Mapping Before Deleting Port
+            dataBase = SonicV2Connector(host="127.0.0.1")
+            if_name_map, if_oid_map = port_util.get_interface_oid_map(dataBase)
+            self.sysLog(syslog.LOG_DEBUG, 'if_name_map {}'.format(if_name_map))
+
+            # If we are here, then get ready to update the Config DB, Update
+            # deletion of Config first, then verify in Asic DB for port deletion,
+            # then update addition of ports in config DB.
+            self.writeConfigDB(delConfigToLoad)
+            # Verify in Asic DB,
+            self._verifyAsicDB(db=dataBase, ports=delPorts, portMap=if_name_map, \
+                timeout=MAX_WAIT)
+            self.writeConfigDB(addConfigtoLoad)
+
+        except Exception as e:
+            print(e)
+            return None, False
+
+        return None, True
+
+    def _deletePorts(self, ports=list(), force=False):
+        '''
+        Delete ports and dependecies from data tree, validate and return resultant
+        config.
+
+        Parameters:
+            ports (list): list of ports
+            force (bool): if false return dependecies, else delete dependencies.
+
+        Returns:
+            (configToLoad, deps, ret) (tuple)[dict, list, bool]: config, dependecies
+            and success/fail.
+        '''
+        configToLoad = None; deps = None
+        try:
+            self.sysLog(msg="delPorts ports:{} force:{}".format(ports, force))
+
+            print('\nStart Port Deletion')
```

Comment:
I will move prints under interactive mode, but I think since we call it from python-click, it will never be interactive. Searching more on this.

Syslogs, I think, we can keep as is, so that less MSGs appear on screen.

Reply:
click does not affect this sys.flags.interactive. You can give a try.

But if you are not interested in every log message to appear on screen, that is a different call.

Even in that case, you could add an optional flag to your syslog method as 'do_print=False'. Wherever you prefer, the log be printed, you can set the flag, instead of duplicating.

Again your call. I am fine either way. Just wanted to reduce lines, where possible.